specs/language/basic.tex

```diff
@@ -289,6 +293,40 @@
 function that returns no value. Any expression can be explicitly converted to
 \texttt{void}.
 
+\Sub{Scalarized Type Compatability}{Basic.types.scalarized}
+
+\p All types \texttt{T} have a \textit{scalarized representation}, \(SR(T)\),
+which is a list of one or more types representing each scalar element of
+\texttt{T}.
+
+\p Scalarized representations are determined as follows:
+\begin{itemize}
+\item The scalarized representation of an array \texttt{T[n]} is \(SR(T_0), ..
+SR(T_n)\).
+
+\item The scalarized representation of a vector \texttt{vector<T,n>} is \(T_0,
+.. T_n\).
+
+\item The scalarized representation of a matrix \texttt{matrix<T,n, m>} is
```

Comment:
Is this always in row major order like in dxc or will have different order for row major and column major?

Reply:
It doesn't matter for this because it is just the types of the elements. Whether you traverse row or column major you still get the same number of elements and the same one type repeated n * m times.